pkg/urbit/noun/events.c

```diff
@@ -937,7 +900,96 @@ u3e_save(void)
   _ce_patch_free(pat_u);
   _ce_patch_delete();
 
-  _ce_backup();
+  c3_c pax_c[8193];
+  snprintf(pax_c, sizeof(pax_c), "%s/.urb/bhk", u3P.dir_c);
+  if ( c3n == u3e_copy(pax_c) ) {
+    fprintf(stderr, "loom: failed to copy snapshot to %s\r\n", pax_c);
+  }
+}
+
+//! @n (1) Attempt to `dir_c`.
+//! @n (2) Attempt to create north image file in `dir_c`.
+//! @n (3) Attempt to create south image file in `dir_c`.
+//! @n (4) Copy north and south image files to `dir_c` from `u3P.dir_c`.
```

Comment:
Same question as elsewhere with regards to separating/indexing these comments.

Reply:
The thought is to mimic the use of footnotes in prose: further elaboration of a point does not belong at the site the point is made, which applies to comments in code just as much as it does for natural language. This way, comments don't break up the flow of the code because we're able to inline trivial `// ([0-9]+)` comments and point the reader elsewhere *should they need further explanation* of a particular line/chunk of code. The indexed list of "footnotes" (headnotes?) also often provides a quick high level overview of the steps that a given function is performing.

Reply:
I agree that comments can be overused and disruptive. I've made efforts to group them together into reasonably sized blocks (more so in hoon, see the types in `sys/arvo.hoon`). But I find this style to be disruptive and ugly; `//! @n (1)` is clearly something that is written for a machine, not for me. What does doxygen do with it? I can't seem to find it in the manual.

Reply:
> I find this style to be disruptive and ugly

I agree it's unconventional and can trigger a negative knee-jerk reaction like this, but the more I've used the convention, the more pleasant I've found it to use. It nicely separates my development process into "code" and "explain"; whenever I write a line/chunk of code that warrants some deeper explanation, I just drop a `([0-9]+)` comment in line and come back after I've finished with my "code thought" to explain. The division between code and the justification for the code is also perfectly separated in this style. If you just want to look at code, you can. If you just want to look at documentation, you can. If you want to look at both, you easily can because the two are clearly connected. The one drawback that I can think of is that it's a new style and so will take getting used to.

> `//! @n (1)` clearly something that is written for a machine, not for me.

Would `//! (1)` be more palatable? `@n` (an alternate form of [`\n`](https://www.doxygen.nl/manual/commands.html#cmdn)) forces a newline in the generated documentation and so is really just a nice-to-have. I'm happy to remove its use if it's distracting in the source code files.